.github/workflows/docker.yml

```diff
@@ -21,19 +21,23 @@ jobs:
       uses: docker/setup-buildx-action@v3
 
     - name: Login to DockerHub
-      id: login
       uses: docker/login-action@v3
-#      continue-on-error: true # Enable to test without docker hub credentials.
       with:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
 
-    - name: Build and publish Docker image
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: boynux/squid-exporter
+
+    - name: Build and push
       uses: docker/build-push-action@v5
       with:
-        context: .
-        push: ${{ steps.login.outcome == 'success' }}
-        tags: ${{ secrets.DOCKER_USERNAME || 'local' }}/squid-exporter:latest,${{ startsWith(github.ref, 'refs/tags/v') && env.rev_tag || '' }}
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
         platforms: linux/amd64,linux/arm64,linux/arm/v7
-      env:
-        rev_tag: ${{ secrets.DOCKER_USERNAME }}/squid-exporter:${{ github.ref_name }} # Construct tag as env var to make tags line easier to read.
+        pull: true
+        push: true
```

Comment:
if login fails, workflow won't reach this step

Reply:
What `pull` is doing here?